dhall/src/Dhall/Diff.hs

```diff
@@ -667,21 +667,20 @@ diffExpression l@(BoolIf {}) r =
     mismatch l r
 diffExpression l r@(BoolIf {}) =
     mismatch l r
-diffExpression (Let asL bL ) (Let asR bR) =
-    enclosed' "" (keyword "in" <> "  ")
-        (Data.List.NonEmpty.zipWith docA asL asR <> pure docB)
+diffExpression l@(Let {}) r@(Let {}) =
+    enclosed' "    " (keyword "in" <> "  ") (docs l r)
   where
-    docA (Binding cL dL eL) (Binding cR dR eR) = align doc
+    docs (Let aL bL cL dL) (Let aR bR cR dR) =
+        Data.List.NonEmpty.cons (align doc) (docs dL dR)
       where
         doc =   keyword "let"
             <>  " "
-            <>  format " " (diffLabel cL cR)
-            <>  format " " (diffMaybe (colon <> " ") diffExpression dL dR)
+            <>  format " " (diffLabel aL aR)
+            <>  format " " (diffMaybe (colon <> " ") diffExpression bL bR)
             <>  equals
             <>  " "
-            <>  diffExpression eL eR
-
-    docB = diffExpression bL bR
+            <>  diffExpression cL cR
+    docs aL aR = pure (diffExpression aL aR)
```

Comment:
The changes in `Dhall.Diff` are just the result of reverting https://github.com/dhall-lang/dhall-haskell/commit/8a5bfaa3b9d1c697d9703f984f3fd866a06c21c0. But I assume that if we use multi-let in pretty-printing, we'd also want to use it in diffs.

I should probably add a test for diffing multi-lets.

Reply:
> I should probably add a test for diffing multi-lets.

This looks tricky as `dhall diff` generates the diff of the _normalized_ expressions – where AFAIK no `let` can appear.

Reply:
Investigating and testing the changes in `Diff` is blocked on https://github.com/dhall-lang/dhall-haskell/issues/1256. If this needs fixes, we can still add them later on – `let`s don't even appear in the usually normalized diffs.